### ISSUE TITLE
client token tests failing on scality ring 7

### ISSUE THREAD

**Issue by phil-davis** [owncloud/files_primary_s3#452]
On 2021-06-12 all CI passed: https://drone.owncloud.com/owncloud/files_primary_s3/2302
On 2021-06-13 the scality ring 7 failures started happening (and the scality ring 8 server was down) https://drone.owncloud.com/owncloud/files_primary_s3/2303
On 2021-06-14 the scality ring 7 failures started happening (and the scality ring 8 server was down) https://drone.owncloud.com/owncloud/files_primary_s3/2304
On 2021-06-15 an early pipeline step hung, so we do not have a result for "scality ring 7/8"
On 2021-06-16 these scality ring 7 failures happened, scality ring 8 passed (there were also other unrelated fails due to core changes for the expiration time, which was later reverted in core)
On 2021-06-17 only these scality ring 7 failures happened - everything else passed https://drone.owncloud.com/owncloud/files_primary_s3/2311
On 2021-06-18 only these scality ring 7 failures happened - everything else passed https://drone.owncloud.com/owncloud/files_primary_s3/2316

So these tests are consistently failing.

https://drone.owncloud.com/owncloud/files_primary_s3/2311/172/18
```
runsh: Total unexpected failed scenarios throughout the test run:
apiAuth/filesAppAuth.feature:18
apiAuth/filesAppAuth.feature:24
apiAuth/webDavAuth.feature:18
```

```
  @smokeTest
  Scenario: access files app with basic token auth                                       # /var/www/owncloud/testrunner/tests/acceptance/features/apiAuth/filesAppAuth.feature:18
    Given a new client token for "Alice" has been generated                              # AuthContext::aNewClientTokenHasBeenGenerated()
      HTTP status code 404 is not the expected value 200
      Failed asserting that 404 matches expected '200'.
    When user "Alice" requests "/index.php/apps/files" with "GET" using basic token auth # AuthContext::userRequestsURLWithUsingBasicTokenAuth()
    Then the HTTP status code should be "200"                                            # FeatureContext::thenTheHTTPStatusCodeShouldBe()

  @smokeTest
  Scenario: access files app with a client token                                               # /var/www/owncloud/testrunner/tests/acceptance/features/apiAuth/filesAppAuth.feature:24
    Given a new client token for "Alice" has been generated                                    # AuthContext::aNewClientTokenHasBeenGenerated()
      HTTP status code 404 is not the expected value 200
      Failed asserting that 404 matches expected '200'.
    When the user requests "/index.php/apps/files" with "GET" using the generated client token # AuthContext::userRequestsURLWithUsingAClientToken()
    Then the HTTP status code should be "200"                                                  # FeatureContext::thenTheHTTPStatusCode

  Scenario: using WebDAV with token auth                                                   # /var/www/owncloud/testrunner/tests/acceptance/features/apiAuth/webDavAuth.feature:18
    Given a new client token for "Alice" has been generated                                # AuthContext::aNewClientTokenHasBeenGenerated()
      HTTP status code 404 is not the expected value 200
      Failed asserting that 404 matches expected '200'.
    When user "Alice" requests "/remote.php/webdav" with "PROPFIND" using basic token auth # AuthContext::userRequestsURLWithUsingBasicTokenAuth()
    Then the HTTP status code should be "207"                               
```

Note: https://drone.owncloud.com/owncloud/files_primary_s3/2311/180/18
Those 3 scenarios all pass on scality ring 8

https://drone.owncloud.com/owncloud/files_primary_s3/2311/173/18
```
runsh: Total unexpected failed scenarios throughout the test run:
apiProvisioning-v1/apiProvisioningUsingAppPassword.feature:11
apiProvisioning-v1/apiProvisioningUsingAppPassword.feature:39
apiProvisioning-v2/apiProvisioningUsingAppPassword.feature:11
apiProvisioning-v2/apiProvisioningUsingAppPassword.feature:39
```

```
  Scenario: admin deletes the user                                                                                 # /var/www/owncloud/testrunner/tests/acceptance/features/apiProvisioning-v1/apiProvisioningUsingAppPassword.feature:11
    Given user "brand-new-user" has been created with default attributes and without skeleton files                # FeatureContext::userHasBeenCreatedWithDefaultAttributesAndWithoutSkeletonFiles()
    And group "brand-new-group" has been created                                                                   # FeatureContext::groupHasBeenCreated()
    And a new client token for the administrator has been generated                                                # AuthContext::aNewClientTokenForTheAdministratorHasBeenGenerated()
      HTTP status code 404 is not the expected value 200
      Failed asserting that 404 matches expected '200'.
    And a new browser session for the administrator has been started                                               # AuthContext::aNewBrowserSessionForTheAdministratorHasBeenStarted()
    And the user has generated a new app password named "my-client"                                                # AuthContext::aNewAppPasswordHasBeenGenerated()
    When the user requests "/ocs/v1.php/cloud/users/brand-new-user" with "DELETE" using the generated app password # AuthContext::userRequestsURLWithUsingAppPassword()
    Then the HTTP status code should be "200"                                                                      # FeatureContext::thenTheHTTPStatusCodeShouldBe()
    And user "brand-new-user" should not exist                                                                     # FeatureContext::userShouldNotExist()

  @smokeTest
  Scenario: normal user gets their own information using the app password                                       # /var/www/owncloud/testrunner/tests/acceptance/features/apiProvisioning-v1/apiProvisioningUsingAppPassword.feature:39
    Given these users have been created with default attributes and without skeleton files:                     # FeatureContext::theseUsersHaveBeenCreatedWithDefaultAttributesAndWithoutSkeletonFiles()
      | username       | displayname |
      | brand-new-user | New User    |
    And a new client token for "brand-new-user" has been generated                                              # AuthContext::aNewClientTokenHasBeenGenerated()
      HTTP status code 404 is not the expected value 200
      Failed asserting that 404 matches expected '200'.
    And a new browser session for "brand-new-user" has been started                                             # AuthContext::aNewBrowserSessionForHasBeenStarted()
    And the user has generated a new app password named "my-client"                                             # AuthContext::aNewAppPasswordHasBeenGenerated()
    When the user requests "/ocs/v1.php/cloud/users/brand-new-user" with "GET" using the generated app password # AuthContext::userRequestsURLWithUsingAppPassword()
    Then the HTTP status code should be "200"                                                                   # FeatureContext::thenTheHTTPStatusCodeShouldBe()
    And the display name returned by the API should be "New User"                                               # FeatureContext::theDisplayNameReturnedByTheApiShouldBe()

  Scenario: admin deletes the user                                                                                 # /var/www/owncloud/testrunner/tests/acceptance/features/apiProvisioning-v2/apiProvisioningUsingAppPassword.feature:11
    Given user "brand-new-user" has been created with default attributes and without skeleton files                # FeatureContext::userHasBeenCreatedWithDefaultAttributesAndWithoutSkeletonFiles()
    And group "brand-new-group" has been created                                                                   # FeatureContext::groupHasBeenCreated()
    And a new client token for the administrator has been generated                                                # AuthContext::aNewClientTokenForTheAdministratorHasBeenGenerated()
      HTTP status code 404 is not the expected value 200
      Failed asserting that 404 matches expected '200'.
    And a new browser session for the administrator has been started                                               # AuthContext::aNewBrowserSessionForTheAdministratorHasBeenStarted()
    And the user has generated a new app password named "my-client"                                                # AuthContext::aNewAppPasswordHasBeenGenerated()
    When the user requests "/ocs/v2.php/cloud/users/brand-new-user" with "DELETE" using the generated app password # AuthContext::userRequestsURLWithUsingAppPassword()
    Then the HTTP status code should be "200"                                                                      # FeatureContext::thenTheHTTPStatusCodeShouldBe()
    And user "brand-new-user" should not exist                                                                     # FeatureContext::userShouldNotExist()

  @smokeTest
  Scenario: normal user gets their own information using the app password                                       # /var/www/owncloud/testrunner/tests/acceptance/features/apiProvisioning-v2/apiProvisioningUsingAppPassword.feature:39
    Given these users have been created with default attributes and without skeleton files:                     # FeatureContext::theseUsersHaveBeenCreatedWithDefaultAttributesAndWithoutSkeletonFiles()
      | username       | displayname |
      | brand-new-user | New User    |
    And a new client token for "brand-new-user" has been generated                                              # AuthContext::aNewClientTokenHasBeenGenerated()
      HTTP status code 404 is not the expected value 200
      Failed asserting that 404 matches expected '200'.
    And a new browser session for "brand-new-user" has been started                                             # AuthContext::aNewBrowserSessionForHasBeenStarted()
    And the user has generated a new app password named "my-client"                                             # AuthContext::aNewAppPasswordHasBeenGenerated()
    When the user requests "/ocs/v2.php/cloud/users/brand-new-user" with "GET" using the generated app password # AuthContext::userRequestsURLWithUsingAppPassword()
    Then the HTTP status code should be "200"                                                                   # FeatureContext::thenTheHTTPStatusCodeShouldBe()
    And the display name returned by the API should be "New User"                                               # FeatureContext::theDisplayNameReturnedByTheApiShouldBe()
```

Note: https://drone.owncloud.com/owncloud/files_primary_s3/2311/181/18
Those 4 scenarios all pass on scality ring 8

**Comment by phil-davis** [comment]
There is nothing useful in the owncloud log. For example, https://drone.owncloud.com/owncloud/files_primary_s3/2316/172/14
```
{"reqId":"8FXrOyHv23KhP0XAjjPJ","level":2,"time":"2021-06-18T00:36:07+00:00","remoteAddr":"192.168.18.9","user":"--","app":"core","method":"DELETE","url":"\/ocs\/v2.php\/apps\/testing\/api\/v1\/opcache","message":"Login failed: 'admin' (Remote IP: '192.168.18.9')"}
{"reqId":"TZzyYI0xg8ghaJEsxNxW","level":2,"time":"2021-06-18T00:36:07+00:00","remoteAddr":"192.168.18.9","user":"--","app":"core","method":"GET","url":"\/index.php\/apps\/files","message":"Login failed: 'Alice' (Remote IP: '192.168.18.9')"}
...
hoping to see some log message about here
...
{"reqId":"KmxefUFsRb65x86zmg3c","level":2,"time":"2021-06-18T00:38:07+00:00","remoteAddr":"192.168.18.9","user":"admin","app":"no app in context","method":"POST","url":"\/ocs\/v1.php\/apps\/testing\/api\/v1\/increasefileid","message":"Inserting dummy entry with fileid bigger than max int of 32 bits for testing"}
{"reqId":"kwG1pnzU8b1FcQVFnmDV","level":2,"time":"2021-06-18T00:38:31+00:00","remoteAddr":"192.168.18.9","user":"--","app":"core","method":"DELETE","url":"\/ocs\/v1.php\/apps\/files_sharing\/api\/v1\/remote_shares\/pending\/123","message":"Login failed: 'another-admin' (Remote IP: '192.168.18.9')"}
```

There is nothing in `owncloud.log` to indicate what went wrong. The response on the `/token/generate` endpoint is a `404` "not found".

**Comment by phil-davis** [comment]
@micbar these failures of "scality ring 7" started happening last weekend, independently of any change that I can think of to core or files_primary_s3. They pass fine with "scality ring 8" and with the regular S3 storages. They are all tests that generate a "client token" using the `token/generate` end-point.

How do we proceed? Ask the "scality ring 7" provider if they changes/updated anything at the end of last week?

**Comment by phil-davis** [comment]
Note: The "scality ring 7" server has been unavailable for the last week or so.
For example: https://drone.owncloud.com/owncloud/files_primary_s3/2337/172/16
```
Error executing "CreateBucket" on "https://s3.isv.scality.com/owncloud-acce  
  ptance-tests-2337-172"; AWS HTTP error: Client error response [url] https:/  
  /s3.isv.scality.com/owncloud-acceptance-tests-2337-172 [status code] 403 [r  
  eason phrase] Forbidden InvalidAccessKeyId (client): The AWS access key Id   
  you provided does not exist in our records. - <?xml version="1.0" encoding=  
  "UTF-8"?><Error><Code>InvalidAccessKeyId</Code><Message>The AWS access key   
  Id you provided does not exist in our records.</Message><Resource></Resourc  
  e><RequestId>deda9455eb40919c86ec</RequestId></Error> 
```

The bucket might have been removed.

@micbar "scality ring 8" is working. Should we still be also testing against "scality ring 7"?

**Comment by micbar** [comment]
> @micbar "scality ring 8" is working. Should we still be also testing against "scality ring 7"?

We need to ask @jojowein

**Comment by micbar** [comment]
@phil-davis Please Remove scality Ring 7

Version 8 has reached LTS

**Comment by phil-davis** [comment]
Fixed in PR #455 - along with other `.drone.star` changes.

**Comment by phil-davis** [comment]
PR #455 has been merged - scality ring 7 pipelines are gone.
